payjoin-cli/src/app.rs

```diff
@@ -429,27 +425,19 @@ impl App {
 
     #[cfg(not(feature = "v2"))]
     fn handle_get_bip21(&self, amount: Option<Amount>) -> Result<Response<Body>, Error> {
+        use payjoin::{PjUriBuilder, Url};
+
         let address = self
             .bitcoind()
             .map_err(|e| Error::Server(e.into()))?
             .get_new_address(None, None)
             .map_err(|e| Error::Server(e.into()))?
             .assume_checked();
-        let uri_string = if let Some(amount) = amount {
-            format!(
-                "{}?amount={}&pj={}",
-                address.to_qr_uri(),
-                amount.to_btc(),
-                self.config.pj_endpoint
-            )
-        } else {
-            format!("{}?pj={}", address.to_qr_uri(), self.config.pj_endpoint)
-        };
-        let uri = payjoin::Uri::try_from(uri_string.clone())
-            .map_err(|_| Error::Server(anyhow!("Could not parse payjoin URI string.").into()))?;
-        let _ = uri.assume_checked(); // we just got it from bitcoind above
+        let pj_endpoint =
+            Url::parse(&self.config.pj_endpoint).map_err(|e| Error::Server(e.into()))?;
```

Comment:
We should probably parse pj_endpoint as a Url in the config setup rather than run into an issue at this point

Reply:
`AppConfig` have `Deserialize` derived and `Url` doesnt have that so its a bit tricky, would create a separate ticket for that as other props need to change as well